File: src/main/java/nodes/statements/ForStatement.java
package nodes;

import nodes.BaseNode;
import nodes.Condition;

import java.util.List;

public class ForStatement extends Statement {
    private Condition condition;
    private Variable variable;
    private IterationChange iterationChange;
    private List<BaseNode> code;

    public ForStatement(Condition condition, Variable variable, IterationChange iterationChange, List<BaseNode> code) {
        this.condition = condition;
        this.variable = variable;
        this.iterationChange = iterationChange;
        this.code = code;
    }

    @Override
    public String toString() {
        StringBuilder builder = new StringBuilder();
        for (BaseNode node : code) {
            builder.append(node.toString()).append(" ");
        }
        return "for(" + variable.toString() + ";" + condition.toString() + ";" + iterationChange.toString() + "){" +
                builder +
                "}";
    }
}

File: src/main/kotlin/nodes/statements/ForStatement.kt
package nodes.statements

import nodes.BaseNode
import nodes.expressions.Condition
import nodes.expressions.IterationChange
import nodes.expressions.Variable

class ForStatement(
    private val condition: Condition,
    private val variable: Variable,
    private val iterationChange: IterationChange,
    private val code: List<BaseNode>
) : Statement() {
    override fun toString(): String {
        val builder = StringBuilder()
        for (node in code) {
            builder.append(node.toString()).append(" ")
        }
        return "for(" + variable.toString() + ";" + condition.toString() + ";" + iterationChange.toString() + "){" +
                builder +
                "}"
    }
}